CONTRIBUTING.md

```diff
@@ -0,0 +1,370 @@
+standard. There are several tools to ensure you abide by this standard.
+
+Alternatively, use [flake8](http://pypi.python.org/pypi/flake8) tool for
+checking the style of your code. Additional standards are outlined on
+the [code style wiki
+page](LINK HERE).
+
+Please try to maintain backward-compatibility. *UV-CDAT* has lots of
+users with lots of existing code, so don't break it if at all possible.
+If you think breakage is required clearly state why as part of the Pull
+Request. Also, be careful when changing method signatures and add
+deprecation warnings where needed.
+
+### Test-driven Development/Writing Code
+*UV-CDAT* is serious about [Test-driven Development
+(TDD)](http://en.wikipedia.org/wiki/Test-driven_development). This
+development process "relies on the repetition of a very short
+development cycle: first the developer writes an (initially failing)
+automated test case that defines a desired improvement or new function,
+then produces the minimum amount of code to pass that test." So, before
+actually writing any code, you should write your tests. Often the test
+can be taken from the original GitHub issue. However, it is always worth
+considering additional use cases and writing corresponding tests.
+
+Adding tests is one of the most common requests after code is pushed to
+*UV-CDAT*. It is worth getting in the habit of writing tests ahead of
+time so this is never an issue.
+
+#### Writing tests
+
+All tests should go into the *tests* subdirectory of the specific
+package. There are probably many examples already there and looking to
+these for inspiration is suggested.
+
+The `testing.checkimage.py` module has special `check_result_image` function
+that make it easier to check whether plot produced after data extraction and
+transformation are equivalent to baseline. For an example see below:
+
+    import cdms2,sys,vcs,sys,os
+    src = sys.argv[1]
+    pth = os.path.join(os.path.dirname(__file__),"..")
+    sys.path.append(pth)
+    import checkimage
+    x = vcs.init()
+    x.drawlogooff() // It is important to disable logo for testing
+    f = cdms2.open(vcs.prefix+"/sample_data/clt.nc")
+    s = f("clt",slice(0,1),squeeze=1)
+    b = x.createboxfill()
+    b.level_1 = .5
+    b.level_2 = 14.5
+    x.plot(s,b,bg = 1)
+
+    fnm = "test_boxfill_lev1_lev2.png"
+
+    x.png(fnm)
+
+    ret = checkimage.check_result_image(fnm,src,checkimage.defaultThreshold)
+    sys.exit(ret)
+
+#### Running the test suite
+
+The tests can then be run directly inside your build tree by typing::
+
+    ctest
+
+To save time and run tests in Pararallel
+
+    ctest -jN
+
+The tests suite is exhaustive and takes around 20 minutes to run. Often
+it is worth running only a subset of tests first around your changes
+before running the entire suite. This is done using one of the following
+constructs:
+
+    ctest -R test-name
+    ctest -R regex*
+
+#### Running the performance test suite
+
+TODO
+
+### Documenting your code
+
+TODO
```

Comment:
yes... maybe group both into a single section "UV-CDAT Coding Practices" or something like that. Basically I think this should come before/near the pep8 bit.